core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/KtorPlugins.kt

```diff
@@ -43,6 +42,7 @@ internal fun HttpClientConfig<*>.addKtorPlugins(
     install(ServerURLVersionRedirectionPlugin.instance)
     install(PreventURLDecodePlugin.instance)
     install(authenticator.instance)
-    install(HttpTimeout) {
+    for (plugin in d2Configuration.networkplugins()) {
+        install(plugin)
```

Comment:
What happens if the plugin needs some configuration, such as `UserAgentPlugin`? Should the SDK accept a configuration block along with the plugin?